pkgs/racket-test/tests/stxparse/test.rkt

```diff
@@ -1059,3 +1059,9 @@
 (convert-compile-time-error
  (syntax-parse #'(1 2 'bar 4 5 'bar 'foo)
    [((~seq (~between x:nat 2 2) ... z) ...+ expr) (void)]))
+
+;; from Laurent Orseau, issue #3603 (1/2021)
+(require syntax/parse/lib/function-header)
+(test-case "skip keywords in invalid-option-placement"
+  (syntax-parse #'(#:a [a 1] #:b b)
+    [a:formals #'(a a.params)]))
```

Comment:
It would be good to add a few other cases, just to make sure, such as mixing keywords and positional arguments like `#'(a #:b [b 1] c #:d d e . rest)`. (Yes, I know it's horrible, but it's still allowed by the grammar of `lambda` and should be checked for backward compatibility.)